android/app/src/main/java/io/mosip/registration_client/api_services/BiometricsDetailsApi.java

```diff
@@ -0,0 +1,553 @@
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.gemalto.jp2.JP2Decoder;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.mosip.biometrics.util.face.FaceBDIR;
+import io.mosip.biometrics.util.finger.FingerBDIR;
+import io.mosip.biometrics.util.iris.IrisBDIR;
+import io.mosip.registration.clientmanager.constant.AuditEvent;
+import io.mosip.registration.clientmanager.constant.Components;
+import io.mosip.registration.clientmanager.constant.Modality;
+import io.mosip.registration.clientmanager.constant.RegistrationConstants;
+import io.mosip.registration.clientmanager.dto.registration.BiometricsDto;
+import io.mosip.registration.clientmanager.dto.registration.RegistrationDto;
+import io.mosip.registration.clientmanager.dto.sbi.CaptureRequest;
+import io.mosip.registration.clientmanager.dto.sbi.DiscoverRequest;
+import io.mosip.registration.clientmanager.exception.BiometricsServiceException;
+import io.mosip.registration.clientmanager.exception.ClientCheckedException;
+import io.mosip.registration.clientmanager.repository.GlobalParamRepository;
+import io.mosip.registration.clientmanager.service.Biometrics095Service;
+import io.mosip.registration.clientmanager.service.RegistrationServiceImpl;
+import io.mosip.registration.clientmanager.spi.AuditManagerService;
+import io.mosip.registration.clientmanager.spi.RegistrationService;
+import io.mosip.registration.keymanager.util.CryptoUtil;
+import io.mosip.registration.packetmanager.cbeffutil.jaxbclasses.SingleType;
+import io.mosip.registration_client.MainActivity;
+import io.mosip.registration_client.model.BiometricsPigeon;
+
+@Singleton
+public class BiometricsDetailsApi implements BiometricsPigeon.BiometricsApi {
+
+    private Activity activity;
+
+    private final AuditManagerService auditManagerService;
+
+    private final ObjectMapper objectMapper;
+
+    private final Biometrics095Service biometricsService;
+
+    private final RegistrationService registrationService;
+
+    private final GlobalParamRepository globalParamRepository;
+
+
+
+    private Modality currentModality;
+
+    private static final String TAG="BiometricsDetailApi";
+
+    private String callbackId;
+
+    private String fieldId;
+    private String purpose;
+    private int capturedAttempts;
+    private int currentAttempt;
+    private int allowedAttempts;
+    private int qualityThreshold;
+    List<BiometricsDto> biometricsDtoList;
+
+    private List<Bitmap> listBitmaps1=new ArrayList<>();
+    private byte[] byteArrayTester;
+    private List<byte[]> listByteArrayTester1=new ArrayList<>();
+    BiometricsPigeon.Result<String> result1;
```

Comment:
can we rename variables appended by 1 to something logical
result1 to something better

@aviral-mishra-hunter @Piyush7034